static/js/user_groups.ts

```diff
@@ -153,3 +153,26 @@ export function get_user_groups_of_user(user_id: number): UserGroup[] {
     );
     return groups_of_user;
 }
+
+export function get_recursive_subgroups(user_group_id: number): Set<number> | undefined {
+    const user_group = user_group_by_id_dict.get(user_group_id);
+    if (user_group === undefined) {
+        blueslip.error(`Could not find user group with ID ${user_group_id}`);
+        return undefined;
+    }
+
+    const subgroup_ids = new Set(user_group.subgroups);
+
+    for (const subgroup_id of subgroup_ids) {
+        const user_group = user_group_by_id_dict.get(subgroup_id);
```

Comment:
We do not need to use another `groups_checked` like set here since `subgroup_ids` is already a set and will not allow adding an already existing element. (Just mentioning here for clarity.)

Reply:
I think the `groups_checked` set might be helpful for reasons noted in my comment above -- making the code more obviously correct.